src/main.rs

```diff
@@ -266,13 +321,20 @@ fn run(opts: &Opts) -> Result<(), Box<dyn Error>> {
             buffer.pop();
         }
 
+        // Safety: Get an immutable borrow of the buffer despite mutable borrows elsewhere
+        // - The references to buffer are created by the regex split.
+        // - References of interest are stored in the `staging` vec of vecs
+        // - At the end of this loop, before clearing the buffer, values from staging are printed
+        // - Staging is then cleared of values to make it safe to reuse
+        let buf: &String = unsafe { mem::transmute(&buffer) };
```

Comment:
Is there a better way to do this than unsafe?

I have an outer vec that I'm reusing that Rust correctly notes has a lifetime longer than the borrows of `&str` here. But I use and clear the values in the vec before clearing the buffer.

Reply:
There is, see https://users.rust-lang.org/t/review-of-unsafe-usage/61520